opm/parser/eclipse/Parser/MessageContainer.cpp

```diff
@@ -25,94 +25,127 @@
 namespace Opm {
 
 
-    void MessageContainer::error(const std::string& msg, const std::string& filename, const int lineno)
-    {
-        std::unique_ptr<Location> loc(new Location{filename, lineno});
-        m_messages.emplace_back(Message{MessageType::MessageTypeEnum::Error, msg, std::move(loc)});
+    void MessageContainer::error( const std::string& msg,
+                                  const std::string& filename,
+                                  const int lineno ) {
+        m_messages.emplace_back( Message {
```

Comment:
Is not one of the reasons to use `emplace_back` that you can drop the explicit construction and just pass the arguments?
```
m_messages.emplace_back(
    MessageType::MessageTypeEnum::Error,
    msg,
    { filename, lineno });
```